internal/prober/multihttp/script.tmpl

```diff
@@ -67,19 +68,24 @@ function assertHeader(headers, name, matcher) {
 
 export default function() {
 	let response;
+	let url;
 	const vars = {};
 
 	{{ range .Entries }}
 	{{- $url := buildUrl .Request }}
+	{{- $urlString := buildUrlString .Request }}
```

Comment:
OK, I went over this several times, and I don't understand it.

buildURL is building a string that can be evaluated as Javascript, but the new version is _not_ including the query.

buildUrlString is building the URL _with_ the query added in.

`$url` is then fed to the request and `$urlString` is used in the assertions.

I see two optoins:

- the assertions generate metrics that use the URL as originally entered by the user
- the assertions generate metrics that use the effective URL that was used to make the request

Reply:
> the assertions generate metrics that use the URL as originally entered by the user

That was sort of my thinking. The reason being:

- It means on the frontend I can query by URL predictably
- It will keep the metrics/logs for an individual step in grouped by the step. If the variable results in a different value every request, and therefore a different URL, there's no way for me to know "this assertion is the one that belongs to the third request", etc